packages/core/src/types.ts

```diff
@@ -821,27 +830,58 @@ export interface HistoryStateNode<TContext> extends StateNode<TContext> {
   target: StateValue | undefined;
 }
 
+// @ts-ignore TS complains about withConfig & withContext not being compatible here when extending StateNode
```

Comment:
this should hopefully get fixed on the next branch since `StateMachine` shouldn't overload `StateNode` like this there